.hlint.yaml

```diff
@@ -0,0 +1,69 @@
+###########################################################################
+# Settings
+###########################################################################
+
+- arguments: [-XTypeApplications, -XRecursiveDo]
+
+# These are just too annoying
+- ignore: { name: Redundant do }
+- ignore: { name: Redundant bracket }
+- ignore: { name: Redundant lambda }
+- ignore: { name: Redundant $ }
+- ignore: { name: Redundant flip }
+- ignore: { name: Move brackets to avoid $ }
+
+# Losing variable names can be not-nice
+- ignore: { name: Eta reduce }
+- ignore: { name: Avoid lambda }
+
+# Humans know better
+- ignore: { name: Use camelCase }
+- ignore: { name: Use const }
+- ignore: { name: Use section }
+- ignore: { name: Use if }
+- ignore: { name: Use notElem }
+- ignore: { name: Use fromMaybe }
+- ignore: { name: Use maybe }
+- ignore: { name: Use fmap }
+- ignore: { name: Use foldl }
+- ignore: { name: "Use :" }
+- ignore: { name: Use ++ }
+- ignore: { name: Use || }
+- ignore: { name: Use && }
+- ignore: { name: 'Use ?~' }
+- ignore: { name: Use <$> }
+
+# Sometimes [Char] is okay (if it means "a set of characters")
```

Comment:
"a set of characters" is best represented by `Set Char`, but `[Char]` can be also interpreted as a sequence of characters (that does not necessarily represent any coherent string), for instance the alphabet.